homeassistant/components/deluge/sensor.py

```diff
@@ -19,14 +19,26 @@
 from homeassistant.helpers.typing import StateType
 
 from . import DelugeEntity
-from .const import CURRENT_STATUS, DATA_KEYS, DOMAIN, DOWNLOAD_SPEED, UPLOAD_SPEED
+from .const import (
+    CURRENT_STATUS,
+    DOMAIN,
+    DOWNLOAD_SPEED,
+    PROTOCOL_TRAFFIC_DOWNLOAD_SPEED,
+    PROTOCOL_TRAFFIC_UPLOAD_SPEED,
+    UPLOAD_SPEED,
+    DelugeGetSessionStatusKeys,
+)
 from .coordinator import DelugeDataUpdateCoordinator
 
 
 def get_state(data: dict[str, float], key: str) -> str | float:
     """Get current download/upload state."""
-    upload = data[DATA_KEYS[0]] - data[DATA_KEYS[2]]
-    download = data[DATA_KEYS[1]] - data[DATA_KEYS[3]]
+    upload = data[DelugeGetSessionStatusKeys.UPLOAD_RATE.value]
+    download = data[DelugeGetSessionStatusKeys.DOWNLOAD_RATE.value]
```

Comment:
This looks to be a breaking change. Previously it was the non dht data rate. Now the sensors report something else.

Reply:
Well, before it was reporting the Upload / Download value, after it was subtracts the DHT (Protocol traffic) upload / download value, which was the inspiration for this PR, as if the up/down rate is 0 and the DHT up/down is > 0, this sensor would report negative

I'm not entirely sure on what constitutes a breaking change as this is my first PR, but if the value representing slightly different information counts , I can mark it as a breaking change

Reply:
i have marked the PR as a breaking change